gateway/src/project.rs

```diff
@@ -748,11 +785,11 @@ impl Service {
         let network = safe_unwrap!(container.network_settings.networks)
             .values()
             .next()
-            .ok_or_else(|| ProjectError::internal("project was not linked to a network"))?;
+            .ok_or_else(|| ProjectError::no_network("project was not linked to a network"))?;
 
         let target = safe_unwrap!(network.ip_address)
             .parse()
-            .map_err(|_| ProjectError::internal("project did not join the network"))?;
+            .map_err(|_| ProjectError::no_network("project did not join the network"))?;
```

Comment:
Nope, not from my testing. So when the network is not joined, then `container.network_settings.networks` will be an empty array (line 785). This would cause the `.next()` (line 787) to fail and we return the correct error in that case.

To truly harden this, we should probably rather loop through all the networks that were joined and find the one we know we created.